scripts.py

```diff
@@ -1269,77 +1267,114 @@ def make_query(self, _db, identifier_type, identifiers, log=None):
+                # it probably means that a human metadata professional had added an explanation/expansion to the 
+                # sort_name, s.a. "Bob A. Jones" --> "Bob A. (Allan) Jones", and we'd rather not replace this data 
+                # with the "Jones, Bob A." that the auto-algorigthm would generate.
+                length_difference = len(contributor.sort_name.strip()) - len(computed_sort_name_local_new.strip())
+                if abs(length_difference) > 3:
+                    cls.register_problem(source="CheckContributorNamesInDB", contribution=contribution, contributor=contributor, 
+                        computed_sort_name=computed_sort_name_local_new, log=log)
+                    output = "%s|\t%s|\t%s|\t%s|\tlength|\tcomplain" % (contributor.id, contributor.sort_name, contributor.display_name, computed_sort_name_local_new, match_ratio)
+                    print output.encode("utf8")
+                    return
+
+                match_ratio = contributor_name_match_ratio(contributor.sort_name, computed_sort_name_local_new, normalize_names=False)
+
+                if (match_ratio < 40):
+                    # ask a human.  this kind of score can happen when the sort_name is a transliteration of the display_name, 
+                    # and is non-trivial to fix.  
+                    cls.register_problem(source="CheckContributorNamesInDB", contribution=contribution, contributor=contributor, 
+                        computed_sort_name=computed_sort_name_local_new, log=log)
+                    output = "%s|\t%s|\t%s|\t%s|\t%s|\tcomplain" % (contributor.id, contributor.sort_name, contributor.display_name, computed_sort_name_local_new, match_ratio)
+                    print output.encode("utf8")
+                else:
+                    # we can fix it!
+                    output = "%s|\t%s|\t%s|\t%s|\t%s|\tfix" % (contributor.id, contributor.sort_name, contributor.display_name, computed_sort_name_local_new, match_ratio)
+                    print output.encode("utf8")
+
+                    contributor.sort_name = computed_sort_name_local_new
 
-        computed_sort_name_local = display_name_to_sort_name(contributor.display_name, advanced=True)
-        success = cls.check_sort_name(computed_sort_name_local, contributor, edition, DataSource.INTERNAL_PROCESSING, CoverageRecord.REPAIR_SORT_NAME_OPERATION)
-        if not success:
-            output = "Contributor[%s]: contributor_sort_name=%s, contributor_display_name=%s: INTERNAL MISMATCH" % (contributor.id, contributor.sort_name, contributor.display_name)
-            print output.encode("utf8")
+                    # also change edition.sort_author, if the author was primary
+                    edition_contributors = contribution.edition.author_contributors
+                    if (len(edition_contributors) > 0 and (edition_contributors[0].id == contributor.id)):
+                        contribution.edition.sort_author = computed_sort_name_local_new
```

Comment:
I would expect calculate_author() to manage Edition.sort_author. Is there a reason why it wouldn't work to set the contributor sort_name and then call Edition.calculate_author()?